addons/survey/models/survey_user_input.py

```diff
@@ -861,7 +878,11 @@ def _get_answer_score_values(self, vals, compute_speed_score=True):
             if answer_type == 'suggestion':
                 suggested_answer_id = vals.get('suggested_answer_id')
                 if suggested_answer_id:
-                    question_answer = self.env['survey.question.answer'].browse(int(suggested_answer_id))
+                    question_answer = (
+                        suggested_answer_id
+                        if isinstance(suggested_answer_id, self.env['survey.question.answer'].__class__)
+                        else self.env['survey.question.answer'].browse(int(suggested_answer_id))
+                    )
```

Comment:
I had missed this commit :upside_down_face: 

To simplify, I would make "_get_answer_score_values" take browserecords for "question", "user_input" and "suggested_answer".

And then in the create you would do:

```py
all_question_ids = [vals.get['question_id'] if vals.get('question_id') for vals in vals_list]
SurveyQuestion = self.env['survey_question'].with_prefetch(all_question_ids)
...
_get_answer_score_values({'question': SurveyQuestion.browse(question_id)})
```
(Hopefully clear, I skipped a few steps).

That way we leverage the power of "with_prefetch" and avoid instance checks that are rather dirty, not a fan of having a method taking either an ID or a browserecord for a param called "_id".

That is of course another approach that requires to change things in master rather than in 17.0.